packages/tools/viewer/src/viewer.ts

```diff
@@ -1497,6 +1909,77 @@ export class Viewer implements IDisposable {
         await Promise.all(promises);
     }
 
+    /**
+     * The current environment skybox mode.
+     * - auto: The default environment is used as the skybox.
+     * - lighting: The environment lighting is used as the skybox.
+     * - url: The specified URL is used as the skybox.
+     */
+    public get environmentSkybox() {

```

Comment:
Why are these new `environmentSkybox` and `environmentLighting` properties required? Can't we just directly update the env lighting as needed when shadows are enabled? If we can avoid adding these properties I think that would be preferable so we don't end up with both functions and properties for loading the env.